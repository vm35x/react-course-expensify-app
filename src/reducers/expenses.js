//Expenses Reducer

const expensesReducerDefaultState = [];

export default (state = expensesReducerDefaultState, action) => {
  //debugger;
  switch (action.type) {
    case "ADD_EXPENSE":
      return [...state, action.expense];
    //return state.concat(action.expense)
    case "REMOVE_EXPENSE":
      return state.filter(({ id }) => id !== action.id);
    case "EDIT_EXPENSE":
      return state.map(expense => {
        if (expense.id === action.id) {
          return {
            ...expense,
            ...action.updates
          };
        } else {
          return expense;
        }
      });
    case "SET_EXPENSES":
      return action.expenses;
    default:
      return state;
  }
};
