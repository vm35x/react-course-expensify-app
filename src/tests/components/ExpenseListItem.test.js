import React from "react";
import { shallow } from "enzyme";
import expenses from "../fixtures/expenses";
import ExpenseListItem from "../../components/ExpenseListItem";

test("should render expense list item on render", () => {
  // const { description, amount, createdAt, id } = expenses[0];
  const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);
  expect(wrapper).toMatchSnapshot();
});

