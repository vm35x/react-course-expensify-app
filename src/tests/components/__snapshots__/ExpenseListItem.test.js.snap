// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list item on render 1`] = `
<Link
  className="list-item"
  replace={false}
  to="edit/1"
>
  <div>
    <h3
      className="list-item__title"
    >
      Rent
    </h3>
    <span
      className="list-item__sub-title"
    >
      December 31st, 1969
    </span>
  </div>
  <h3
    className="list-item__data"
  >
    $11.95
  </h3>
</Link>
`;
